src/shared/tpm2-util.h

```diff
@@ -67,18 +88,31 @@ static inline void Esys_Freep(void *p) {
                 sym_Esys_Free(*(void**) p);
 }
 
-int tpm2_get_good_pcr_banks(ESYS_CONTEXT *c, uint32_t pcr_mask, TPMI_ALG_HASH **ret_banks);
-
-#else
-struct tpm2_context;
-#endif
-
-int tpm2_context_init(const char *device, struct tpm2_context *ret);
-void tpm2_context_destroy(struct tpm2_context *c);
+int tpm2_get_good_pcr_banks(struct tpm2_context *c, uint32_t pcr_mask, TPMI_ALG_HASH **ret_banks);
 
 int tpm2_list_devices(void);
 int tpm2_find_device_auto(int log_level, char **ret);
 
+#else /* HAVE_TPM2 */
+
+struct tpm2_context {};
```

Comment:
hmm, why define it to an empty struct? it should suffic to define this one without declaring any actual definition, no, as before?

Reply:
ack, changed